components/renku_data_services/k8s/models.py

```diff
@@ -161,6 +161,30 @@ def to_v1_secret(self) -> V1Secret:
             type=self.manifest.get("type"),
         )
 
+    def to_patch(self) -> list[dict[str, Any]]:
+        """Create a rfc6902 patch that would take an existing secret and patch it to this state."""
+        if self.manifest.get("stringData"):
+            raise NotImplementedError("Patching a secret with stringData field is not implemented.")
+        patch = [
+            {"op": "replace", "path": "/data", "value": self.manifest.data},
+            {"op": "replace", "path": "/type", "value": self.manifest.get("type", "Opaque")},
+        ]
+        if "metadata" not in self.manifest:
+            return patch
+        if "labels" in self.manifest.metadata:
+            patch.append(
+                {"op": "replace", "path": "/metadata/labels", "value": self.manifest.metadata.labels},
+            )
+        if "annotations" in self.manifest.metadata:
+            patch.append(
+                {"op": "replace", "path": "/metadata/annotations", "value": self.manifest.metadata.annotations},
+            )
+        if "ownerReferences" in self.manifest.metadata:
+            patch.append(
+                {"op": "replace", "path": "/metadata/ownerReferences", "value": self.manifest.metadata.ownerReferences},
+            )
+        return patch
```

Comment:
```suggestion
        # We never create 'finalizers' nor 'managedFields', so we do not patch them.
        return patch
```

Reply:
done in https://github.com/SwissDataScienceCenter/renku-data-services/pull/1063/commits/4ffba7bea2c90b9fc71472abeb75bb341d8f8d36